tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -329,20 +374,49 @@ private void AssertAuthenticationFails(MongoClientSettings settings)
         {
             return CreateDatabaseUser(client, "admin", userName, password, "readWriteAnyDatabase", mechanisms);
         }
-        
+
         private BsonDocument CreateDatabaseUser(
             MongoClient client,
             string source,
-            string userName, 
-            string password, 
-            string role, 
+            string userName,
+            string password,
+            string role,
             params string[] mechanisms)
         {
             var createUserCommand =
-                $"{{createUser: '{userName}', pwd: '{password}'," 
+                $"{{createUser: '{userName}', pwd: '{password}',"
                 + $"roles: ['{role}'], mechanisms: {mechanisms.ToJson()}}}";
             return client.GetDatabase(source).RunCommand<BsonDocument>(createUserCommand);
         }
-        
+
+        private BsonDocument CreateX509DatabaseUser(MongoClient client, string userName)
+        {
+            var createUserCommand =
+                $"{{ createUser : '{userName}'," +
+                @"   roles : [ { role : 'readWrite', db : 'test' },
+                               { role : 'userAdminAnyDatabase', db : 'admin' } ]
+                  }";
+            return client.GetDatabase("$external").RunCommand<BsonDocument>(createUserCommand);
+        }
+
+        private void DropDatabaseUser(MongoClient client, string userName, string database)
```

Comment:
A more logical order for the parameters is:

    private void DropDatabaseUser(MongoClient client, string database, string userName)